src/TConsole.cpp

```diff
@@ -2253,6 +2253,34 @@ TLabel* TConsole::createLabel(const QString& name, int x, int y, int width, int
     }
 }
 
+std::pair<bool, QString> TConsole::deleteLabel(const QString& name)
+{
+    if (name.isEmpty()) {
+        return {false, QLatin1String("a label cannot have an empty string as its name")};
+    }
+
+    auto pL = mLabelMap.take(name);
+    if (pL) {
+        // Using deleteLater() rather than delete as it seems a safer option
+        // given that this item is likely to be linked to some events and
+        // suchlike:
+        pL->deleteLater();
+
+        // It remains to be seen if the label has "gone" as a result of the
+        // above by the time the Lua subsystem processes the following:
+        TEvent mudletEvent{};
+        mudletEvent.mArgumentList.append(QLatin1String("sysLabelDeletion"));
```

Comment:
I think `sysLabelDeleted` is a better name - I know, I know technically it's deleted whenever Qt decides to, but as far as the Lua system is concerned it'll be deleted and no longer usable (nor on the screen!).
```suggestion
        mudletEvent.mArgumentList.append(QLatin1String("sysLabelDeleted"));
```